services/storage/tests/test_rest.py

```diff
@@ -313,9 +312,15 @@ async def _fake_download_to_file_or_raise(session, url, dest_path):
     assert dsm
     assert isinstance(dsm, DataStorageManager)
 
-    mock = mocker.patch.object(dsm, "download_link_datcore")
-    mock.return_value = Future()
-    mock.return_value.set_result(("https://httpbin.org/image", "foo.txt"))
+    async def mock_download_link_datcore(*args, **kwargs):
+        return ["https://httpbin.org/image", "foo.txt"]
+
+    mocker.patch.object(dsm, "download_link_datcore", mock_download_link_datcore)
+    # FIXME: mock fails in py3.8 It await dsm.download_link_datcore(...) -> returns a mock instead of the results.
+    # it is a matter of knowning how to replace the member function in the dsm object
+    # examples show that with a class object instead of an instance so probably we are
+    # using the mocker wrong here.
+    # mock.return_value.set_result(["https://httpbin.org/image", "foo.txt"])
```

Comment:
or maybe we could use aioresponses?

Reply:
This is actually an old comment since the mock is fixed. 
Regarding aioreponses. Yes, the idea is to introduce something uniform in the storage refactoring backlog #2276